zerver/migrations/0493_rename_userhotspot_to_onboardingstep.py

```diff
@@ -0,0 +1,129 @@
+                        raw_query = SQL(
+                            "ALTER TABLE {table_name} DROP CONSTRAINT {old_name}"
+                        ).format(table_name=Identifier(old_table), old_name=Identifier(old_name))
+                    cursor.execute(raw_query)
+                    continue
+
+                seen_indexes.add(new_name)
+                if is_index:
+                    raw_query = SQL("ALTER INDEX {old_name} RENAME TO {new_name}").format(
+                        old_name=Identifier(old_name), new_name=Identifier(new_name)
+                    )
+                else:
+                    raw_query = SQL(
+                        "ALTER TABLE {old_table} RENAME CONSTRAINT {old_name} TO {new_name}"
+                    ).format(
+                        old_table=Identifier(old_table),
+                        old_name=Identifier(old_name),
+                        new_name=Identifier(new_name),
+                    )
+                cursor.execute(raw_query)
+
+            for infodict in connection.introspection.get_sequences(cursor, old_table):
+                old_name = infodict["name"]
+                column = infodict["column"]
+                new_name = f"{new_table}_{column}_seq"
+
+                raw_query = SQL("ALTER SEQUENCE {old_name} RENAME TO {new_name}").format(
+                    old_name=Identifier(old_name),
+                    new_name=Identifier(new_name),
+                )
+                cursor.execute(raw_query)
+
+            cursor.execute(
+                SQL("ALTER TABLE {old_table} RENAME TO {new_table}").format(
+                    old_table=Identifier(old_table), new_table=Identifier(new_table)
+                )
+            )
+
+    return inner_migration
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("zerver", "0492_realm_push_notifications_enabled_and_more"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunPython(
+                    rename_indexes_constraints_column_and_table(
+                        "zerver_userhotspot",
+                        "zerver_onboardingstep",
+                        old_column="hotspot",
+                        new_column="onboarding_step",
+                    ),
+                    reverse_code=rename_indexes_constraints_column_and_table(
+                        "zerver_onboardingstep",
+                        "zerver_userhotspot",
+                        old_column="onboarding_step",
+                        new_column="hotspot",
+                    ),
+                )
+            ],
+            state_operations=[
+                migrations.RenameField(
+                    model_name="UserHotspot", old_name="hotspot", new_name="onboarding_step"
+                ),
+                migrations.RenameModel(old_name="UserHotspot", new_name="OnboardingStep"),
+            ],
+        ),
+    ]
```

Comment:
@alexmv @mateuszmandera can you review the migration code here? I didn't closely follow the similar work in the previous rename migration so am not sure how to verify whether this rather tangly logic is correct.

Reply:
I have moved the function to `lib/migrate` with a slight tweak in comments to make it generic. (2nd commit)